### ISSUE TITLE
Don't recommend insecure usage of keys

### ISSUE THREAD

**Issue by joshribakoff** [serverless/serverless#11763]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

https://www.serverless.com/framework/docs/providers/aws/guide/credentials/#using-aws-access-keys

You instruct users to run:

```
serverless config credentials \
  --provider aws \
  --key AKIAIOSFODNN7EXAMPLE \
  --secret wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
```

passing secrets in the CLI is not secure! This logs the secret in plaintext in the users bash/zsh history! Instead, the command should be interactive taking in the input via stdin interactively so it's not logged into the shell's history.

I classified this as a bug because you're telling users to do something that is widely regarded as a bad security practice, and then you're referring to that as "the permanent" solution implying that it is secure or suitable for production environments.

### Service configuration (serverless.yml) content

```yaml
n/a
```

### Command name and used flags

n/a

### Command output

```shell
n/a
```

### Environment information

```shell
n/a
```

**Comment by medikoo** [comment]
@joshribakoff I agree. I think `serverless config credentials` command should be deprecated and removed in future major release of the Framework. For AWS creds CLI setup, developers should be advised to simply use `aws configure`

@skierkowski what do you think?

**Comment by skierkowski** [comment]
Deprecating the `serverless config credentials` command is fine; however, depending on the `aws` CLI command would make the barrier to entry for new Serverless Framework users higher as they'd also have to install the `aws` CLI and follow the AWS docs. We can remove the command, but I'd recommend that we instruct users to set the credentials in the `~/.aws/credentials` file manually as the first option; using the `aws` CLI can be a secondary option.

**Comment by medikoo** [comment]
So I believe we can start with following steps:
1. Improve https://www.serverless.com/framework/docs/providers/aws/guide/credentials/ documentation:
    1. Remove `setup with serverless config credentials command` section 
    2. Remove `setup with the aws-cli`
    3. Improve the other sections wording in light of above changes
2. Configure deprecation for `config credentials`

**Comment by duytrandev** [comment]
Hi @medikoo 

Can I propose running `serverless config credentials --provider aws` would simply trigger AWS CLI `aws configure`.
AWS CLI already allowed users to key in secret key in a secured manner, why not just simply letting user options to use both Serverless CLI and AWS CLI.

If no one is working on this issue, I can pick this up.

Thanks,

**Comment by medikoo** [comment]
> Can I propose running serverless config credentials --provider aws would simply trigger AWS CLI aws configure.
AWS CLI already allowed users to key in secret key in a secured manner, why not just simply letting user options to use both Serverless CLI and AWS CLI.

@duytrandev I'm not sure if we should run another CLI behind the scenes (we don't have guarantee that AWS CLI is available in the system).  It's ok to advise users to rely on this command, in deprecation message.

So I'd say let's stick to https://github.com/serverless/serverless/issues/11763#issuecomment-1446926589, and PR for that is welcome!

**Comment by duytrandev** [comment]
@medikoo how about we simulate the behavior of `aws configure`, this can both remove the dependency on AWS CLI and provide a CLI for user to setup their credentials?

**Comment by medikoo** [comment]
> @medikoo how about we simulate the behavior of aws configure, 

Wouldn't that just add extra maintenance cost? What would be the reason to re-implement what AWS CLI does in Framework CLI?

**Comment by duytrandev** [comment]
Maybe not the full implementation of `aws configure` function, but just a part of it asking for Access Key ID and Secret Key.
It would be an enhancement to the current way of working that allows users to secure their secrets.

Keeping the cli `serverless configs credentials` would greatly help CICD pipelines & automated jobs to set up and run without the need for the extra aws cli, or having to manipulate the file system in the process (which can be quite cumbersome doing by scripts)

**Comment by medikoo** [comment]
> Keeping the cli serverless configs credentials would greatly help CICD pipelines 

@duytrandev I think in the interactive form, it'll be hard to use in CICD pipelines, so I'm not sure if this use case is valid (?)

**Comment by joshribakoff** [comment]
If this framework doesn’t depend on AWS CLI, and instead uses its own implementation, then it should also implement secure credential configuration. Either that, or depend on AWS CLI. Seems like this shouldn’t be so difficult. If the framework is going to “reimplement” aws cli, then it should weigh into that decision and provide a secure setup command.

**Comment by selfsimilar** [comment]
This seems related to https://github.com/serverless/serverless/issues/10181 and to my own issue where I'm currently forced to pass the AWS access key and secret access key via CLI, when I would rather use an AWS Role or OIDC, which would be more secure than keeping long-lived static access tokens in my CI/CD pipeline.
